src/extensions/wcf/Microsoft.DotNet.UpgradeAssistant.Extensions.WCFUpdater/WCFUpdateStep.cs

```diff
@@ -210,29 +211,71 @@ public Task<UpgradeStepApplyResult> Apply()
 
         private void FindPath(IProject project)
         {
-            var csFile = project.FindFiles(".cs", ProjectItemType.Compile);
-            var main = from f in csFile
-                       where File.Exists(f) && File.ReadAllText(f).Replace(" ", string.Empty).IndexOf("Main(", StringComparison.Ordinal) >= 0
-                       select f;
-            var directives = from f in csFile
-                             where File.Exists(f) && File.ReadAllText(f).IndexOf("using System.ServiceModel", StringComparison.Ordinal) >= 0
-                             && File.ReadAllText(f).Replace(" ", string.Empty).IndexOf("Main(", StringComparison.Ordinal) < 0
-                             select f;
-            var config = from f in project.FindFiles(".config", ProjectItemType.None)
-                         where File.Exists(f) && File.ReadAllText(f).IndexOf("<system.serviceModel>", StringComparison.Ordinal) >= 0
-                         select f;
-
-            if (!main.Any())
+            var compileItems = project.FindFiles(".cs", ProjectItemType.Compile);
+            var configItems = project.FindFiles(".config", ProjectItemType.None);
```

Comment:
Other flavors link windows etc you conditioned and skipped. Why WCF changes were needed ? Are you expecting them for MacOS? I assume you tried it out on Windows for WCF to verify?

Reply:
I actually modified this code originally in order to get all of the unit tests passing on macOS and then later realized that we didn't care about supporting WCF on macOS.

Honestly all this change really does (as big as it might look) is to unwind the LINQ statements in order to do path conversions and to eliminate duplication of File.ReadAllText() that happens 3 times per file in the \*.cs case (1x in the `main = ` statement and 2x in the `directives =` statement).

Yes, the changes have been verified on Windows.